lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3868,10 +3724,9 @@ void TypeChecker::validateDecl(ValueDecl *D) {
 
     DeclValidationRAII IBV(FD);
 
-    // Bind operator functions to the corresponding operator declaration.
-    if (FD->isOperator())
-      bindFuncDeclToOperator(*this, FD);
-
+    // Force computing the operator decl in case it emits diagnostics.
```

Comment:
We do request forcing in typeCheckDecl(). No need to trigger this for decls in secondary files for SILGen's sake.

Also further down in the same switch case there is this block: `// Member functions need some special validation logic.`. I think this check should go into the request or even just typeCheckDecl() itself.